x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -4023,6 +4024,25 @@ public void testRejectRemoteEnrichAfterCoordinatorEnrich() {
         );
     }
 
+    public void testNoStackOverflow() {
+        String query = randomBoolean() ? "row a = 1" : "row a = 1 | eval b = a";
+        for (int i = 0; i < 1900; i++) {
+            query += "::long::int";
+        }
+        for (int i = 0; i < 50; i++) {
+            for (int j = 0; j < randomInt(20); j++) {
+                query += "::long::int";
+            }
```

Comment:
What is this trying to do? Trigger one of the two types of `ParsingException`s?

Reply:
See my comment above. Depending on the order of tests, the behavior can change, so the only way to try to catch the StackOverflow at parsing time vs at planning time is to randomize and iterate.